File: cmake_template/hello.cpp
#include "hello.hpp"

double hello_cpp(int number, double real_number) {

  std::cout << "Hello from C++" << std::endl;
  std::cout << "You gave me the integer: " << number << std::endl;
  std::cout << "You gave me the double: " << real_number << std::endl;

  return number + real_number;
}

File: cmake_template/hello.f90


SUBROUTINE hello_fortran(number, real_number, total_number)
  IMPLICIT NONE
  INTEGER, INTENT(in)  :: number !input
  REAL, INTENT(in) :: real_number !input
  REAL, INTENT(out) :: total_number ! output

  WRITE(*,*) "Hello from Fortran"
  WRITE(*,*) "You gave me the integer: ", number
  WRITE(*,*) "You gave me the float: ", real_number

  total_number = number + real_number

END SUBROUTINE hello_fortran
